File: Minimum-Viable-Project_MVP/templates/lang_transl.js
$(document).ready(function () {
    // Populate language dropdowns
    $.each(LANGUAGES, function (code, language) {
        $('#source-lang').append($('<option>').val(code).text(language));
        $('#target-lang').append($('<option>').val(code).text(language));
    });

    // Translate button click event
    $('#translate-btn').click(function () {
        var text = $('#text-input').val();
        var sourceLang = $('#source-lang').val();
        var targetLang = $('#target-lang').val();

        if (text && sourceLang && targetLang) {
            // Translate text
            $.ajax({
                url: '/translate',
                type: 'POST',
                dataType: 'json',
                data: {
                    text: text,
                    sourceLang: sourceLang,
                    targetLang: targetLang
                },
                success: function (response) {
                    $('#output').text(response.translatedText);
                },
                error: function () {
                    alert('An error occurred during translation.');
                }
            });
        } else {
            alert('Please enter text and select source/target languages.');
        }
    });
});

File: Minimum-Viable-Project_MVP/templates/lang_transl.py
#!/usr/bin/python3

from flask import Flask, request, render_template, jsonify
from googletrans import LANGUAGES, Translator

app = Flask(__name__)

@app.route('/')
def index():
    return render_template('index.html')

@app.route('/translate', methods=['POST'])
def translate():
    text = request.form['text']
    source_lang = request.form['source']
    target_lang = request.form['target']

    if text and source_lang and target_lang:
        # Create a translator instance
        translator = Translator(service_urls=['translate.google.com'])

        # Translate the text
        result = translator.translate(text, src=source_lang, dest=target_lang)

        # Return the translated text as JSON response
        return jsonify({'translatedText': result.text})

    return jsonify({'error': 'Missing text, source language, or target language.'})

if __name__ == '__main__':
    app.run(debug=True)

